presto-main/src/main/java/com/facebook/presto/operator/aggregation/state/TwoNullableValueStateMapping.java

```diff
@@ -63,19 +66,62 @@ private TwoNullableValueStateMapping() {}
                 .put(ImmutableList.of(Block.class, Slice.class), BlockSliceState.class)
                 .put(ImmutableList.of(Block.class, void.class), BlockUnknownState.class)
                 .put(ImmutableList.of(void.class, void.class), TwoNullableValueState.class)
-                .put(ImmutableList.of(void.class, Block.class), UnknownBlockState.class)
+                .put(ImmutableList.of(void.class, Block.class), BlockPositionValueState.class)
                 .put(ImmutableList.of(void.class, boolean.class), UnknownBooleanState.class)
                 .put(ImmutableList.of(void.class, double.class), UnknownDoubleState.class)
                 .put(ImmutableList.of(void.class, long.class), UnknownLongState.class)
-                .put(ImmutableList.of(void.class, Slice.class), UnknownSliceState.class)
+                .put(ImmutableList.of(void.class, Slice.class), BlockPositionValueState.class)
                 .build();
     }
 
     public static Class<? extends AccumulatorState> getStateClass(Class<?> first, Class<?> second)
     {
         List<Class<?>> key = ImmutableList.of(first, second);
-        Class<? extends AccumulatorState> state = MAPPINGS.get(key);
+        Class<? extends AccumulatorState> state = STATE_MAPPINGS.get(key);
         checkArgument(state != null, "Unsupported state type combination: (%s, %s)", first.getName(), second.getName());
         return state;
     }
+
+    public static AccumulatorStateSerializer<?> getStateSerializer(Type firstType, Type secondType)
+    {
+        List<Class<?>> key = ImmutableList.of(firstType.getJavaType(), secondType.getJavaType());
+
+        if (key.equals(ImmutableList.of(boolean.class, Block.class))) {
```

Comment:
Why don't we just put into an ImmutableMap and initialized in static scope? (Similar to `STATE_MAPPINGS`)